Http connector throws when trying to get response

**Describe the bug**

Starting with version 1.1.0, I get the following error when trying to access the response of the http request from an http connector:

```
Unexpected exception thrown: org.operaton.connect.ConnectorResponseException: HTCL-02004 Unable to read connectorResponse: Stream already closed
        at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:152)
        at org.junit.jupiter.api.AssertDoesNotThrow.createAssertionFailedError(AssertDoesNotThrow.java:84)
        at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:75)
        at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:58)
        at org.junit.jupiter.api.Assertions.assertDoesNotThrow(Assertions.java:3259)
        at repro.HttpConnectorReproTest. httpConnector_getRequest_shouldHaveReadableResponse(HttpConnectorReproTest.java:24)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: org.operaton.connect.ConnectorResponseException: HTCL-02004 Unable to read connectorResponse: Stream already closed
        at org.operaton.connect.httpclient.impl.HttpConnectorLogger.unableToReadResponse(HttpConnectorLogger.java:40)
        at org.operaton.connect.httpclient.impl.HttpResponseImpl.collectResponseParameters(HttpResponseImpl.java:76)
        at org.operaton.connect.impl.AbstractConnectorResponse.getResponseParameters(AbstractConnectorResponse.java:36)
        at org.operaton.connect.impl.AbstractConnectorResponse.getResponseParameter(AbstractConnectorResponse.java:44)
        at org.operaton.connect.httpclient.impl.HttpResponseImpl.getResponse(HttpResponseImpl.java:48)
        at repro.HttpConnectorReproTest.lambda$httpConnector_getRequest_shouldHaveReadableResponse $0(HttpConnectorReproTest.java:24)
        at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:71)
        ... 6 more
Caused by: connectjar.org.apache.hc.core5.http.StreamClosedException: Stream already closed
        at connectjar.org.apache.hc.core5.http.impl.io.ContentLengthInputStream.read(ContentLengthInputStream.java:162)
        at connectjar.org.apache.hc.core5.http.io.EofSensorInputStream.read(EofSensorInputStream.java:138)
        at java.base/java.io.InputStream.readNBytes(InputStream.java:409)
        at java.base/java.io.InputStream.readAllBytes(InputStream.java:346)
        at org.operaton.connect.httpclient.impl.HttpResponseImpl.collectResponseParameters(HttpResponseImpl.java:73)
        ... 11 more
```

When downgrading to version 1.0.3, everything works as expected (=> no error).

**To Reproduce**

While I first saw this error when running it via a workflow, I was able to reproduce it by directly using the `HttpConnectorImpl` with the following, very simple test:

```java
@Test
void httpConnector_getRequest_shouldHaveReadableResponse() {
  HttpConnectorImpl connector = new HttpConnectorImpl();

  HttpRequest request = connector.createRequest();
  request.setRequestParameter("method", "GET");
  request.setRequestParameter("url", "https://httpbin.org/get");

  HttpResponse response = request.execute();

  assertDoesNotThrow(() -> response.getResponse());
}
```

--- a/commons/utils/src/main/java/org/operaton/commons/utils/IoUtil.java
+++ b/commons/utils/src/main/java/org/operaton/commons/utils/IoUtil.java
@@ -49,7 +49,7 @@ public class IoUtil {
    * @return the input stream as byte[].
    */
   public static byte[] inputStreamAsByteArray(InputStream inputStream) {
-    try {
+    try (inputStream) {
       return inputStream.readAllBytes();
     } catch (IOException e) {
       throw LOG.unableToReadInputStream(e);

--- a/connect/http-client/src/main/java/org/operaton/connect/httpclient/impl/HttpRequestInvocation.java
+++ b/connect/http-client/src/main/java/org/operaton/connect/httpclient/impl/HttpRequestInvocation.java
@@ -19,6 +19,10 @@ package org.operaton.connect.httpclient.impl;
 import java.util.List;
 
 import org.apache.hc.client5.http.classic.HttpClient;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.io.entity.ByteArrayEntity;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
 import org.apache.hc.core5.http.message.BasicClassicHttpRequest;
 
 import org.operaton.connect.impl.AbstractRequestInvocation;
@@ -39,7 +43,17 @@ public class HttpRequestInvocation extends AbstractRequestInvocation<BasicClassi
 
   @Override
   public Object invokeTarget() throws Exception {
-    return client.execute(target, response -> response);
+    return client.execute(target, response -> {
+      // Buffer the response entity so it remains readable after the connection is closed.
+      // HttpClient 5 closes the underlying stream once this handler returns.
+      HttpEntity entity = response.getEntity();
+      if (entity != null) {
+        response.setEntity(new ByteArrayEntity(
+            EntityUtils.toByteArray(entity),
+            ContentType.parseLenient(entity.getContentType())));
+      }
+      return response;
+    });
   }
 
 }

--- a/connect/http-client/src/main/java/org/operaton/connect/httpclient/impl/HttpResponseImpl.java
+++ b/connect/http-client/src/main/java/org/operaton/connect/httpclient/impl/HttpResponseImpl.java
@@ -18,6 +18,7 @@ package org.operaton.connect.httpclient.impl;
 
 import java.io.Closeable;
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -70,7 +71,7 @@ public class HttpResponseImpl extends AbstractCloseableConnectorResponse impleme
 
     if (httpResponse.getEntity() != null) {
       try {
-        String response = new String(httpResponse.getEntity().getContent().readAllBytes());
+        String response = new String(httpResponse.getEntity().getContent().readAllBytes(), StandardCharsets.UTF_8);
         responseParameters.put(PARAM_NAME_RESPONSE, response);
       } catch (IOException e) {
         throw LOG.unableToReadResponse(e);

--- a/connect/http-client/src/test/java/org/operaton/connect/httpclient/impl/HttpRequestInvocationTest.java
+++ b/connect/http-client/src/test/java/org/operaton/connect/httpclient/impl/HttpRequestInvocationTest.java
@@ -15,17 +15,24 @@
  */
 package org.operaton.connect.httpclient.impl;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
 
 import org.apache.hc.client5.http.classic.HttpClient;
 import org.apache.hc.client5.http.classic.methods.HttpGet;
 import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
 import org.apache.hc.core5.http.io.HttpClientResponseHandler;
+import org.apache.hc.core5.http.io.entity.InputStreamEntity;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 
+import org.operaton.connect.httpclient.TestResponse;
 import org.operaton.connect.spi.ConnectorRequest;
 import org.operaton.connect.spi.ConnectorRequestInterceptor;
 
@@ -158,5 +165,67 @@ class HttpRequestInvocationTest {
 
     // then
     assertThat(request).isEqualTo(connectorRequest);
+  }
+
+  /**
+   * Regression test for <a href="https://github.com/operaton/operaton/issues/2607">#2607</a>.
+   * <p>
+   * Simulates real HttpClient 5 behavior: after {@code execute(request, handler)} returns,
+   * the underlying response entity stream is closed. The response body must still be readable
+   * because {@link HttpResponseImpl#collectResponseParameters} reads it lazily.
+   */
+  @Test
+  void invokeTarget_shouldReturnReadableResponse_whenStreamClosedAfterHandler() throws Exception {
+    // given
+    String expectedBody = "Hello, World!";
+    byte[] bodyBytes = expectedBody.getBytes(StandardCharsets.UTF_8);
+
+    when(httpClient.execute(eq(httpRequest), any(HttpClientResponseHandler.class)))
+        .thenAnswer(invocation -> {
+          HttpClientResponseHandler<?> handler = invocation.getArgument(1);
+
+          // An InputStream that throws after close — just like the real
+          // ContentLengthInputStream in HttpClient 5
+          InputStream entityStream = new InputStream() {
+            private final ByteArrayInputStream delegate = new ByteArrayInputStream(bodyBytes);
+            private boolean closed = false;
+
+            @Override
+            public int read() throws IOException {
+              if (closed) throw new IOException("Stream already closed");
+              return delegate.read();
+            }
+
+            @Override
+            public int read(byte[] b, int off, int len) throws IOException {
+              if (closed) throw new IOException("Stream already closed");
+              return delegate.read(b, off, len);
+            }
+
+            @Override
+            public void close() {
+              closed = true;
+            }
+          };
+
+          TestResponse response = new TestResponse();
+          response.setEntity(
+              new InputStreamEntity(entityStream, bodyBytes.length, ContentType.TEXT_PLAIN));
+
+          // HttpClient 5 calls the handler, then closes the underlying stream
+          Object result = handler.handleResponse(response);
+          entityStream.close();
+          return result;
+        });
+
+    HttpRequestInvocation inv = new HttpRequestInvocation(
+        httpRequest, connectorRequest, interceptorChain, httpClient);
+
+    // when — simulate what AbstractHttpConnector.execute() does after invokeTarget()
+    ClassicHttpResponse httpResponse = (ClassicHttpResponse) inv.invokeTarget();
+    HttpResponseImpl responseImpl = new HttpResponseImpl(httpResponse);
+
+    // then — reading the response body must still work
+    assertThat(responseImpl.getResponse()).isEqualTo(expectedBody);
   }
 }

